Core/MemMap.cpp

```diff
@@ -457,4 +477,77 @@ void Memset(const u32 _Address, const u8 _iValue, const u32 _iLength) {
 	CBreakPoints::ExecMemCheck(_Address, true, _iLength, currentMIPS->pc);
 }
 
+bool HandleFault(uintptr_t hostAddress, void *ctx) {
+	SContext *context = (SContext *)ctx;
+	const uint8_t *codePtr = (uint8_t *)(context->CTX_PC);
+
+	// TODO: Check that codePtr is within the current JIT space.
+	bool inJitSpace = MIPSComp::jit->CodeInRange(codePtr);
+	if (!inJitSpace) {
+		// This is a crash in non-jitted code. Not something we want to handle here, ignore.
+		return false;
+	}
+
+	uintptr_t baseAddress = (uintptr_t)base;
+#ifdef MASKED_PSP_MEMORY
+	const uintptr_t addressSpaceSize = 0x40000000ULL;
+#else
+	const uintptr_t addressSpaceSize = 0x100000000ULL;
+#endif
+
+	// Check whether hostAddress is within the PSP memory space, which (likely) means it was a guest executable that did the bad access.
+	if (hostAddress < baseAddress || hostAddress >= baseAddress + addressSpaceSize) {
+		// Host address outside - this was a different kind of crash.
+		return false;
+	}
+
+	// OK, a guest executable did a bad access. Take care of it.
+
+	uint32_t guestAddress = hostAddress - baseAddress;
+
+	// TODO: Share the struct between the various analyzers, that will allow us to share most of
+	// the implementations here.
+
+#if defined(PPSSPP_ARCH_AMD64) || defined(PPSSPP_ARCH_X86)
+	// X86, X86-64. Variable instruction size so need to analyze the mov instruction in detail.
+
+	// To ignore the access, we need to disassemble the instruction and modify context->CTX_PC
+	LSInstructionInfo info;
+	X86AnalyzeMOV(codePtr, info);
+#elif defined(PPSSPP_ARCH_ARM64)
+	uint32_t word;
+	memcpy(&word, codePtr, 4);
+	// To ignore the access, we need to disassemble the instruction and modify context->CTX_PC
+	Arm64LSInstructionInfo info;
+	Arm64AnalyzeLoadStore((uint64_t)codePtr, word, &info);
+#elif defined(PPSSPP_ARCH_ARM)
+	uint32_t word;
+	memcpy(&word, codePtr, 4);
+	// To ignore the access, we need to disassemble the instruction and modify context->CTX_PC
+	ArmLSInstructionInfo info;
+	ArmAnalyzeLoadStore((uint32_t)codePtr, word, &info);
+#endif
+
+	if (g_Config.bIgnoreBadMemAccess) {
+		if (!info.isMemoryWrite) {
+			// It was a read. Fill the destination register with 0.
+			// TODO
+		}
+		// Move on to the next instruction.
+		context->CTX_PC += info.instructionSize;
+		// Fall through to logging.
+	} else {
+		// Jump to a crash handler that will exit the game.
+		context->CTX_PC = (uintptr_t)MIPSComp::jit->GetCrashHandler();
+		ERROR_LOG(SYSTEM, "Bad memory access detected! %08x (%p) Stopping emulation.", guestAddress, (void *)hostAddress);
+		return true;
+	}
+
+	g_numReportedBadAccesses++;
+	if (g_numReportedBadAccesses < 100) {
+		ERROR_LOG(SYSTEM, "Bad memory access detected and ignored: %08x (%p)", guestAddress, (void *)hostAddress);
```

Comment:
Would be good to match the logging/reporting for non-fast memory.  Should probably be using `MEMMAP` instead of `SYSTEM` for these.

-[Unknown]

Reply:
Changed it to MEMMAP at least, text still isn't quite the same though.